scripts/version-linter.py

```diff
@@ -24,26 +24,58 @@
 import sys
 import re
 import sexpdata
+import requests
 
 exit_code=0
 
 def set_error():
   global exit_code
   exit_code=1
 
-def branch_commit(branch):
+def latest_branch_commit(branch):
   print ('Retrieving', branch, 'head commit...')
   result=subprocess.run(['git','log','-n','1','--format="%h"','--abbrev=7',f'{branch}'],
                         capture_output=True)
   output=result.stdout.decode('ascii')
   print ('command stdout:', output)
   print ('command stderr:', result.stderr.decode('ascii'))
-  return output.replace('"','').replace('\n','')
-
-def download_type_shapes(role,branch,sha1) :
+  return output.replace('"','').splitlines()
+
+def url_to_type_shape_file(file):
+  '''
+    Return url to mina type shape file
+  '''
+  return f'https://storage.googleapis.com/mina-type-shapes/{file}'
+
+def sha_exists(sha1):
+  '''
+    Checks if mina type shape with given sha exists
+  '''
+  file = type_shape_file(sha1)
+  return url_exists(url_to_type_shape_file(file))
+
+def url_exists(url):
+  '''
+    Checks if url exists (by sending head and validating that status code is ok)
+  '''
+  return requests.head(url).status_code == 200
+
+def find_latest_type_shape_ref_on(branch):
```

Comment:
This function is redundant now that `latest_branch_commit` only returns 1 commit. Recommend deleting this and the helper functions